### ISSUE TITLE
Parca Agent drops data unexpectedly?

### ISSUE THREAD

**Issue by brancz** [parca-dev/parca-agent#1563]
**Describe the bug**

I don't have enough information yet to be 100% sure that this is a Parca Agent issue, but over time I've noticed that sometimes data that should be available for querying is not available, but it starts to show up if I restart the Parca Agent.

For example, here we see that processes with the `comm="cockroach"` label suddenly disappear even though the processes have existed throughout the whole timeframe:

<img width="1820" alt="Screenshot 2023-04-17 at 15 45 36" src="https://user-images.githubusercontent.com/4546722/232502899-69f53442-405c-4640-9ae2-94e76ee3d1a2.png">

I restarted the parca-agent on the node of the disappeared line and data started to appear again.

**To Reproduce**
Steps to reproduce the behavior: Run Parca Agent for an extended time and search for various processes that you know are actively using CPU time.

**Expected behavior**

**Screenshots** or **Profiles** (using https://pprof.me)

What's curious is that data 

<img width="1920" alt="Screenshot 2023-04-17 at 15 42 30" src="https://user-images.githubusercontent.com/4546722/232502647-036a52cc-4ef5-4ac5-81dc-0407a9549529.png">

**Logs**
If applicable, could you add logs to help explain your problem?

I'll try to obtain some next time I notice this.

**Software (please complete the following information):**
 - Parca Agent Version: v0.17.1
 - Parca Server Version (if applicable): Polar Signals Cloud

**Workload (please complete the following information):**
happens seemingly randomly, not related to runtime or compiler, hard to say though since we have almost only Go workloads

**Environment (please complete the following information):**
- Linux Distribution: [`cat /etc/*-release`] Google container optimized Linux
- Linux Version: [`uname -r`] GKE 1.24 (whatever that is)
- Arch: [`uname -m`] amd64
- Kubernetes Version (if applicable): [`kubectl --version`] GKE 1.24
- Container Runtime (if applicable): containerd

**Additional context**

We'll make sure to get some more metrics from the Parca Agents themselves, maybe they will be able to tell us if more or less data is sent over time.

**Comment by asubiotto** [comment]
Based on some internal metrics, it looks like the disappearance of the series is correlated with a process restart. When running the parca agent with debug logging enabled and searching for the new process PID, a ton of these log lines are observed:
```
2023-04-19 11:42:03 | level=debug name=parca-agent ts=2023-04-19T11:42:03.087167035Z caller=cpu.go:952 msg="failed to normalize address" pid=193163 address=4994817 err="mapping is nil"
```
Internal metrics also corroborate the fact that normalization failures shoot up after a restart. Given that a failure to normalize means we drop a stack trace
https://github.com/parca-dev/parca-agent/blob/55183f02755c6f99e09cd95fb44fd48eebc0bee4/pkg/profiler/cpu/cpu.go#L971-L975
The likely reason for the disappearance of the series is these normalization failures.

**Comment by thorfour** [comment]
To further corroborate this theory I was able to recreate this behavior by performing a rolling restart of a deployment and validate that some series went missing.

**Comment by kakkoyun** [comment]
Fixed by #1507
